src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -187,6 +191,7 @@ public String createSnapshot(
       // We fetch the DUOS dataset to confirm its existence, but do not need the returned value.
       duosClient.getDataset(duosId, userReq);
     }
+    validateForByRequestIdMode(snapshotRequestModel.getContents().get(0));
```

Comment:
So here is my thinking. Currently, this method first checks to ensure the mode is byRequestId and returns right away if it is not. It only does validation for the byRequestId mode. I think it might be beneficial to the reader for the method to have a more descriptive name that reflects what it currently does. If this is updated later to be a more broad validation step, then the name should be updated as well. Thoughts?

Reply:
It's hard to know what will be helpful for a future reader. My philosophy is to provide less detail whenever possible, because -- for me -- detail makes it harder to think in generalities, and -- also for me -- I find it easier to understand code when I can think generally about it.